### PR TITLE
Add Created, CreatedBy, and CompletedBy to a Branch in the model cache.

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Add Created, CreatedBy, and CompletedBy to a Branch in the model cache,
multiwatcher, allwatcher and modelcache worker. Including helper
functions to get the data from the model cache.

## QA steps

Not directly used yet.  Unit tests should all pass and no difference in juju functionality found.

### CODE REVIEW

File: core/cache/branch.go [hunk 1]
```diff
@@ -58,6 +58,28 @@ func (b *Branch) AppConfig(appName string) settings.ItemChanges {
...
+}
+
+// CreatedBy returns user who created the branch.
+func (b *Branch) CreatedBy() string {
+	return b.details.CreatedBy
+}
+
+// Completed returns a Unix timestamp indicating when this generation
+// was committed.
+func (b *Branch) Completed() int64 {
```
Comment (manadart) [inline core/cache/branch.go]: Completed branches are removed from the cache.
With the working design, branches in the cache will have the zero value for `Completed` and `CompletedBy`.

There is no harm in adding these properties, but are they needed for status?

Comment (hmlanigan) [reply core/cache/branch.go]: They are needed for 'juju commits # alias list-commits', per the spec.  Per agreement, I'll leave it in, doesn't have to be used right now.

**Comment by hmlanigan** [comment]
force push to resolve spelling error caught by go linter.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**